### ISSUE TITLE
Latest version breaks with RequireJS (global is undefined)

### ISSUE THREAD

**Issue by boskee** [wycats/handlebars.js#1004]
Hello,

It looks like the latest version breaks when used with RequireJS (specifically handlebars.amd.js and handlebars.amd.min.js)

The problem exists in the following piece of code (starting at line 3859 in handlebars.amd.js):

```javascript
  var $Handlebars = global.Handlebars;
  /* istanbul ignore next */
  inst.noConflict = function () {
    if (global.Handlebars === inst) {
      global.Handlebars = $Handlebars;
    }
  };
```

And it can be fixed by using the snippet starting at line 724 of the same file:

```javascript
  var root = typeof global !== 'undefined' ? global : window,
      $Handlebars = root.Handlebars;
  /* istanbul ignore next */
  Handlebars.noConflict = function () {
    if (root.Handlebars === Handlebars) {
      root.Handlebars = $Handlebars;
    }
  };
```

So the code in question should become:

```javascript
  var root = typeof global !== 'undefined' ? global : window,
      $Handlebars = root.Handlebars;
  /* istanbul ignore next */
  inst.noConflict = function () {
    if (root.Handlebars === inst) {
      root.Handlebars = $Handlebars;
    }
  };
```

**Comment by kpdecker** [comment]
Sorry for this. This is a pretty simple fix and I hope to release a patch for this soon.

**Comment by kpdecker** [comment]
Released in 3.0.3
